primary/src/tests/helper_tests.rs

```diff
@@ -147,23 +209,38 @@ async fn test_process_certificates_batch_mode() {
     }
 
     // AND spin up a mock node
-    let address = committee.primary(&name).unwrap();
-    let mut handler = PrimaryToPrimaryMockServer::spawn(address.primary_to_primary);
+    let address = committee
+        .primary(&requestor_name)
+        .unwrap()
+        .primary_to_primary;
+    let requestor_key = primary_keys.pop().unwrap();
+    let (mut handler, _network) = PrimaryToPrimaryMockServer::spawn(requestor_key, address);
+
+    // Wait for connectivity
```

Comment:
Wondering if this is worth factoring out to a common test util? It seems to appear in several places here & other files

Reply:
I think there's actually quite a bit of work that would be done to refactor lots of the common testing boilerplate we currently have. Given this is a large change already (and there's more to come) it might make sense to wait till the dust settles so we have a better base to refactor from.

Reply:
SGTM!